libcodechecker/analyze/analyzers/result_handler_plist_to_db.py

```diff
@@ -182,43 +192,53 @@ def handle_results(self):
          - addFileContent
          - finishBuildAction
         """
-
-        with client.get_connection() as connection:
-
-            LOG.debug('Storing original build and analyzer command '
-                      'to the database.')
-
-            _, source_file_name = os.path.split(self.analyzed_source_file)
-
-            if LoggerFactory.get_log_level() == logger.DEBUG:
-                analyzer_cmd = ' '.join(self.analyzer_cmd)
-            else:
-                analyzer_cmd = ''
-
-            build_cmd_hash = self.buildaction.original_command_hash
-            analysis_id = \
-                connection.add_build_action(self.__run_id,
-                                            build_cmd_hash,
-                                            analyzer_cmd,
-                                            self.buildaction.analyzer_type,
-                                            source_file_name)
-
-            assert self.analyzer_returncode == 0
-
-            plist_file = self.analyzer_result_file
-
-            try:
-                files, reports = plist_parser.parse_plist(plist_file)
-            except Exception as ex:
-                LOG.debug(str(ex))
-                msg = 'Parsing the generated result file failed.'
-                LOG.error(msg + ' ' + plist_file)
-                connection.finish_build_action(analysis_id, msg)
-                return 1
-
-            self.__store_bugs(files, reports, connection, analysis_id)
-
-            connection.finish_build_action(analysis_id, self.analyzer_stderr)
+        if not client:
+            LOG.error("Client needs to be set to store the "
+                      "results to the database")
+            return
+
+        LOG.debug('Storing original build and analyzer command '
+                  'to the database.')
+
+        _, source_file_name = os.path.split(self.analyzed_source_file)
+
+        if LoggerFactory.get_log_level() == logger.DEBUG:
+            analyzer_cmd = ' '.join(self.analyzer_cmd)
+        else:
+            analyzer_cmd = ''
+
+        build_cmd_hash = self.buildaction.original_command_hash
+        analysis_id = \
+            client.addBuildAction(self.__run_id,
+                                  build_cmd_hash,
+                                  analyzer_cmd,
+                                  self.buildaction.analyzer_type,
+                                  source_file_name)
+
+        assert self.analyzer_returncode == 0
```

Comment:
Maybe assert is not the best way to handle the error here. The return code is not zero is not due to a programming error. It might be a configuration error.